artifactory/commands/transferfiles/transfer.go

```diff
@@ -192,7 +194,12 @@ func (tdc *TransferFilesCommand) transferSingleRepo(sourceRepoKey string, target
 		if tdc.ShouldStop() {
 			return
 		}
-
+		// Ensure the data structure which stores the upload tasks on Artifactory's side is wiped clean,
+		// in case some of the requests to delete handles tasks sent by JFrog CLI did not reach Artifactory.
+		err = stopTransferInArtifactory(tdc.sourceServerDetails, srcUpService)
```

Comment:
I think it would be better to put in a defer before the "for" loop. Something like:
```go
defer func ()  {
	if tdc.ShouldStop() {
		// stopTransferInArtifactory already invoked
		return
	}
	stopTransferInArtifactory(tdc.sourceServerDetails, srcUpService)
	if err != nil {
		log.Error(err)
	}
}
```